### PR TITLE
Add documentation for the http client response `success?` method

### PR DESCRIPTION
**Pull request by marceloboeira**

### CODE REVIEW

File: src/http/client/response.cr [hunk 1]
```diff
@@ -31,6 +31,7 @@ class HTTP::Client::Response
     @body
   end
 
+  # Returns true if the client's request was successfully received, understood, and accepted.
```
Comment (asterite) [inline src/http/client/response.cr]: I'd prefer to be explicit here and document that `success?` means that the response status code is in the range `200..299`.

Comment (marceloboeira) [inline src/http/client/response.cr]: Alright!

**Comment by marceloboeira** [comment]
@asterite updated!

**Comment by asterite** [comment]
Thanks!

**Pull request merged**